cryostat/templates/NOTES.txt

```diff
@@ -0,0 +1,84 @@
+{{- $envVars = list "CRYOSTAT_WEB_HOST=127.0.0.1" "CRYOSTAT_EXT_WEB_PORT=8080" }}
+{{- $portForwards = prepend $portForwards "8080:$CONTAINER_PORT" }}
+{{- end }}
+
+{{- if .Values.core.route.enabled }}
+  export GRAFANA_ROUTE_HOST=$(oc get route -n {{ .Release.Namespace }} {{ include "cryostat.fullname" . }}-grafana -o jsonpath="{.status.ingress[0].host}")
+{{- $envVars = append $envVars ( tpl "GRAFANA_DASHBOARD_URL=http{{ if .Values.grafana.route.tls.enabled }}s{{ end }}://$GRAFANA_ROUTE_HOST" . ) }}
+{{- else if .Values.grafana.ingress.enabled }}
+{{- /* Do nothing */}}
+{{- else if contains "NodePort" .Values.grafana.service.type }}
+{{- if not (contains "NodePort" .Values.core.service.type) }}
+  export NODE_IP=$(kubectl get nodes -n {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+{{- end }}
+  export GRAFANA_NODE_PORT=$(kubectl get -n {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "cryostat.fullname" . }}-grafana)
+{{- $envVars = append $envVars "GRAFANA_DASHBOARD_URL=http://$NODE_IP:$GRAFANA_NODE_PORT"}}
+{{- else if contains "LoadBalancer" .Values.grafana.service.type }}
+{{- if not (contains "LoadBalancer" .Values.core.service.type) }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get -n {{ .Release.Namespace }} -w svc/{{ include "cryostat.fullname" . }} svc/{{ include "cryostat.fullname" . }}-grafana'
+{{- end }}
+  export GRAFANA_SERVICE_IP=$(kubectl get svc -n {{ .Release.Namespace }} {{ include "cryostat.fullname" . }}-grafana --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+{{- $envVars = append $envVars (printf "GRAFANA_DASHBOARD_URL=http://$GRAFANA_SERVICE_IP:%v" .Values.grafana.service.port) }}
+{{- else if contains "ClusterIP" .Values.grafana.service.type }}
+{{- if not (contains "ClusterIP" .Values.core.service.type) }}
+  export POD_NAME=$(kubectl get pods -n {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "cryostat.name" . }},app.kubernetes.io/instance={{ .Release.Name }}"  --sort-by=.metadata.creationTimestamp -o jsonpath="{.items[-1:].metadata.name}")
+{{- end }}
+  export GRAFANA_CONTAINER_PORT=$(kubectl get pod -n {{ .Release.Namespace }} $POD_NAME -o jsonpath="{.spec.containers[1].ports[0].containerPort}")
+{{- $envVars = concat $envVars ( list "GRAFANA_DASHBOARD_URL=http://127.0.0.1:$GRAFANA_CONTAINER_PORT" "GRAFANA_DASHBOARD_EXT_URL=http://127.0.0.1:8081" )}}
+{{- $portForwards = append $portForwards "8081:$GRAFANA_CONTAINER_PORT" }}
+{{- end }}
+
+{{- if not (empty $envVars) }}
+  kubectl -n {{ .Release.Namespace }} set env deploy --containers={{ .Chart.Name }} {{ include "cryostat.fullname" . }} {{ join " " $envVars }}
+{{- end }}
+
+{{- if not (empty $portForwards) }}
+
+{{ $listNum }}. Forward local ports to the application's pod:
```

Comment:
I tried running these steps from the `NOTES` that were printed and ran into an error on this `kubectl port-forward` step because I had done it too quickly and my Pod wasn't running yet. Would it make sense to add a step here with an ex. `kubectl wait --for=condition=available --timeout=60s` to wait for the deployment to be stable/ready before moving on?

Reply:
Good suggestion! I've done this now